#159291136 Validate all cases of user's entries via Testing

phemmelliot opened the pull request:
#### What does this PR do?
    In this PR i wrote test for all the endpoints, the test was written with mocha 
   and chai and the test has support for es6. It has babel compilers add to the test script.
#### List of Task to be completed?
    1. installed mocha, chai and chai-http dependencies for testing
    2. Imported the dependencies into test file and wrote the test for all
        5 endpoints covering as much edge cases as possible.
    3. Included babel compilers in the test script 
    4. I added node_modules to gitignore
#### How should this be manually tested?
     This test can be carried out by using npm test
#### Any background context you want to provide?
     I was initially using jasmine to write my test, but it turns out to be a little hard 
     for me so changed to mocha and chai, then i was able to test my endpoints easily
#### What are the relevant pivotal tracker stories?
     Here is the link to 
   [pivotal tracker board](https://www.pivotaltracker.com/story/show/159291136)
#### Screenshot
     
![test](https://user-images.githubusercontent.com/19429611/43260862-95b70e48-90d2-11e8-8fd6-5b5e2e060124.png)

phemmelliot committed:
feature(testing): Tests all endpoints
 -Make sure as many as possible edge cases are covered
 [Delivers #159291136]

andela-ookoro commented on node_modules/rimraf/package.json:
```diff
@@ -1,27 +1,29 @@
 {
-  "_from": "rimraf@^2.2.8",
+  "_from": "rimraf",
```
why did you post this file

phemmelliot replied:
When i was going through babel issues yesterday, rimraf was added to build script for performing a unix command rm -rf on a folder, but it turns out to be an ineffective approach so i dumped, but i didn't know it was pushed, next i am pushing i will add node_modules to gitignore

andela-ookoro commented on package.json:
```diff
@@ -26,18 +27,31 @@
+    "chai-http": "^4.0.0",
     "express": "^4.16.3",
+    "mocha": "^5.2.0",
+    "morgan": "^1.9.0",
```
what did you use `morgan` for

phemmelliot replied:
I was initially using a tutorial where morgan was used apparently for logging switching between config files of test and live mongo db, but because i did not use any config file, i didn't use morgan

andela-ookoro commented on test/login.spec.js:
```diff
@@ -0,0 +1,142 @@
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import { server, entries } from '../server';
```
nice test

phemmelliot replied:
Thank you

phemmelliot committed:
Added node_modules to gitignore

phemmelliot committed:
Bug: Added node-modules to gitignore

 - Added node_modules to gitignore
 - Removed Unused dependencies from package.json

 [Deliver #159333892]

phemmelliot committed:
Bug: Added node-modules to gitignore again

 - Added node_modules to gitignore

 [Deliver #159333892]

phemmelliot committed:
Bug: Added node-modules to gitignore yet again

 - Added node_modules to gitignore

 [Deliver #159333892]

phemmelliot committed:
Bug: Added node_modules to gitignore

- removed unused dependencies

[Delivers #159333892]

phemmelliot committed:
Revert "Bug: Added node-modules to gitignore"

This reverts commit a1f72f759f1221a4419723c42ec612b85382ef85.

phemmelliot committed:
Bug: Added node-modules to gitignore yet again

 - Revert pushing of node_modules

 [Deliver #159333892]

andela-ookoro reviewed (approved):
this is good
